src/viam/sdk/robot/client.cpp

```diff
@@ -357,11 +362,11 @@ RobotClient::RobotClient(std::shared_ptr<ViamChannel> channel)
     : viam_channel_(channel),
       channel_(channel->channel()),
       should_close_channel_(false),
-      impl_(std::make_unique<Impl>(RobotService::NewStub(channel_))) {}
+      impl_(std::make_unique<impl>(RobotService::NewStub(channel_))) {}
 
-std::vector<Name>* RobotClient::resource_names() {
+std::vector<Name> RobotClient::resource_names() {
     const std::lock_guard<std::mutex> lock(lock_);
-    std::vector<Name>* resources = std::move(&resource_names_);
+    std::vector<Name> resources = std::move(resource_names_);
```

Comment:
I think this is actually going to destroy the contents of `resource_names_` since they will be moved into `resources` and then returned. I think you can get rid of this line entirely and now write `return resource_names_` which will implicitly copy. It actually wouldn't have been possible to make this error if `RobotClient::resource_names` was declared `const`, which I think it probably should be. Can you please make it `const`? Note that you will probably need to declare `lock_` as mutable to do so. That's fine, it is very common for private `mutex` members to be mutable: https://stackoverflow.com/questions/14131572/always-declare-stdmutex-as-mutable-in-c11

Reply:
The `mutable` should go on the declaration of `RobotClient::lock_`.